src/doc/rgw/multisite-reshard.md

```diff
@@ -72,10 +72,19 @@ The distinction between *index layout* and *log layout* is important, because in
 * Full sync uses a single bucket-wide listing to fetch all objects.
     - Use a cls_lock to prevent different shards from duplicating this work.
 * When incremental sync gets to the end of a log shard (i.e. listing the log returns truncated=false):
-    - If the remote has a newer log generation, flag that shard as 'resharded' in the bucket sync status.
-    - Once all shards in the current generation reach that 'resharded' state, incremental bucket sync can advance to the next generation.
+    - If the remote has a newer log generation, flag that shard as 'done' in the bucket sync status.
```

Comment:
The reason is not clear to me, why 'done' instead of 'resharded'?

Reply:
the bucket-wide sync status object will be tracking the current log generation it's working on, along with the set of bucket shards that are finished syncing that generation and ready to transition to the next

i called the state 'resharded' before, because a bucket reshard was the only reason we'd create a new log generation. but this `bucket sync enable` case is a new one, unrelated to resharding

in code, i imagine it looking something like this:
```diff
 struct rgw_bucket_sync_status {
   BucketSyncState state = BucketSyncState::Init;
   rgw_bucket_full_sync_status full;
+  uint64_t incremental_gen;
+  std::vector<bool> shards_done_with_gen;
```